contracts/src/GetWethValueInPoolBatchRequest.sol

```diff
@@ -256,7 +256,7 @@ contract GetWethValueInPoolBatchRequest {
             ///@notice The first value is a Q96 representation of p_token0, the second is 128X fixed point representation of p_token1.
             uint256 priceSquaredShiftQ96 = token0IsReserve0
                 ? priceSquaredX96 / Q96
-                : (Q96 * 0xffffffffffffffffffffffffffffffff) / (priceSquaredX96 / Q96);
+                : (Q96 * 0xffffffffffffffffffffffffffffffff) / (priceSquaredX96 << 128 / Q96);
```

Comment:
You need to change the numerator here. You are trying to invert the price. Your denominator is already Q128 after doing `<< 128`, so you want to do 1/ price. Since it's already in Q128 I believe this should be 

```solidity
uint256 priceSquaredShiftQ96 = token0IsReserve0
                ? priceSquaredX96 / Q96
                : type(uint256).max / (priceSquaredX96 << 128 / Q96);
```

Reply:
Ah yes, my bad. Thanks!